splunk-quarkus/src/main/java/com/redhat/console/notifications/splunkintegration/GenericIntegration.java

```diff
@@ -0,0 +1,170 @@
+        .log(LoggingLevel.ERROR, "IllegalArgumentException for event ${header.ce-id} (account ${header.accountId})"
+            + " to ${header.targetUrl}: ${exception.message}")
+        .log(LoggingLevel.DEBUG, "${exception.stacktrace}")
+        .setBody(simple("${exception.message}"))
+        .setHeader("outcome-fail", simple("true"))
+        .process(resultTransformer)
+        .marshal().json()
+        .process(ceEncoder)
+        .to(direct("return"));
+  }
+
+  private void configureIoFailed() throws Exception {
+    Processor ceEncoder = new CloudEventEncoder(COMPONENT_NAME, RETURN_TYPE);
+    Processor resultTransformer = new ResultTransformer();
+    // The error handler found an IO Exception. We set the outcome to fail and then
+    // send to kafka
+    from(direct("ioFailed"))
+        .routeId("ioFailed")
+        .log(LoggingLevel.ERROR, "IOFailure for event ${header.ce-id} (account ${header.accountId})"
+            + " to ${header.targetUrl}: ${exception.message}")
+        .log(LoggingLevel.DEBUG, "${exception.stacktrace}")
+        .setBody(simple("${exception.message}"))
+        .setHeader("outcome-fail", simple("true"))
+        .process(resultTransformer)
+        .marshal().json()
+        .process(ceEncoder)
+        .to(direct("return"));
+  }
+
+  private void configureHttpFailed() throws Exception {
+    Processor ceEncoder = new CloudEventEncoder(COMPONENT_NAME, RETURN_TYPE);
+    Processor resultTransformer = new ResultTransformer();
+    // The error handler found an HTTP Exception. We set the outcome to fail and
+    // then send to kafka
+    from(direct("httpFailed"))
+        .routeId("httpFailed")
+        .log(LoggingLevel.ERROR, "HTTPFailure for event ${header.ce-id} (account ${header.accountId})"
+            + " to ${header.targetUrl}: ${exception.getStatusCode()}"
+            + " ${exception.getStatusText()}: ${exception.message}")
+        .log(LoggingLevel.DEBUG, "Response Body: ${exception.getResponseBody()}")
+        .log(LoggingLevel.DEBUG, "Response Headers: ${exception.getResponseHeaders()}")
+        .setBody(simple("${exception.message}"))
+        .setHeader("outcome-fail", simple("true"))
+        .process(resultTransformer)
+        .marshal().json()
+        .process(ceEncoder)
+        .to(direct("return"));
+  }
+
+  private void configureIngress() throws Exception {
+    from(kafka(kafkaIngressTopic).brokers(kafkaBrokers).groupId(kafkaIngressGroupId))
+        .routeId("ingress")
+        // Decode CloudEvent
+        .process(new CloudEventDecoder())
+        // We check that this is our type.
+        // Otherwise, we ignore the message there will be another component that takes
+        // care
+        .log("IM WORKING HERE")
+        .filter().simple("${header.ce-type} == '" + CE_TYPE + "'")
+        // Log the parsed cloudevent message.
+        .to(log("com.redhat.console.notifications.splunkintegration?level=DEBUG"))
+        .to(direct("handler"))
```

Comment:
@vkrizan I'm not able to find a way to redirect to the child `handler` route

Do you have any idea?

Thanks